calico_containers/calicoctl.py

```diff
@@ -356,6 +356,39 @@ def node(ip, node_image, log_dir, ip6="", as_num=None):
     # Print warnings for any known system issues before continuing
     checksystem(fix=False, quit_if_error=False)
 
+    # Get IP address of host, if none was specified
+    if not ip:
+        ip = get_first_ip()
+        if ip:
+            print "No IP provided. Using detected IP: %s" % ip
+        else:
+            print "Couldn't autodetect a management IP address. Please rerun" \
+                  "command with the --ip=<IP_ADDRESS> flag."
+            sys.exit(1)
+
```

Comment:
This doesn't do what https://github.com/Metaswitch/calico-docker/issues/233 suggested, which is check that the address specified on the command line exists.